pyro/poutine/subsample_messenger.py

```diff
@@ -49,8 +49,7 @@ def sample(self, sample_shape=torch.Size()):
         if subsample_size is None or subsample_size >= self.size:
             result = torch.arange(self.size, device=self.device)
         else:
-            result = torch.multinomial(torch.ones(self.size), self.subsample_size,
-                                       replacement=False).to(self.device)
+            result = torch.randperm(self.size, device=self.device)[:subsample_size].clone()
```

Comment:
why clone?

Reply:
The `.clone()` frees the `size` chunk of memory. Without the `.clone()` you'd get a `subsample_size`-sized view into a `size`-sized chunk of memory, preventing that memory from being freed.